lib/validator.js

```diff
@@ -0,0 +1,117 @@
+'use strict';
+
+// let validator = module.exports = {};
+
+class Validator {
+  constructor(input, rules) {
+    this.input = input;
+    this.rules = rules;
+  }
+  isValid = () => {
+    return true;
+  };
+  isString = () => {
+    return typeof input === 'string';
+  };
+  isNumber = () => {
+    return typeof input === 'number';
+  };
+
+  isArray = () => {
+    let isInputOfType = array.isArray(input);
+    let isValueOfType = true;
+
+    if (valueType) {
+      for (var i = input.length - 1; i >= 0; i--) {
+        if (typeof input[i] === valueType)
+          continue;
+        else
+          isValueOfType = false;
+      }
+    }
+    return isInputOfType && isValueOfType;
+  }
+  isObject = (input) => {
+    return typeof input === 'object';
+  };
+  isBoolean = (input) => {
+    return typeof input === 'boolean';
+  };
+  isNumber = (input) => {
+    return typeof input === 'number';
+  };
+  isFunction = (input) => {
+    return typeof input === 'function';
+  };
+  isTruthy = (input) => {
+    return !!input;
+  };
+}
+
```

Comment:
for this lab if you can put the class and the constructor into two different files. It can be helpfully this way because you compare the code side-by-side rather  then going up-and-down the screen.